example/mailto/mailto_grammar.hpp

```diff
@@ -0,0 +1,466 @@
+            boost::urls::grammar::range_rule(
+                boost::urls::grammar::tuple_rule(
+                    boost::urls::grammar::optional_rule(fws_rule),
+                    comment_rule), 1),
+            boost::urls::grammar::optional_rule(fws_rule)),
+        fws_rule);
+
+/// Rule for qcontent = qtext / quoted-pair
+constexpr auto qcontent_rule =
+    boost::urls::grammar::variant_rule(
+        boost::urls::grammar::delim_rule(qtext_chars),
+        quoted_pair_rule);
+
+/// Rule for quoted-string = [CFWS] DQUOTE *([FWS] qcontent) [FWS] DQUOTE [CFWS]
+constexpr auto quoted_string_rule =
+    boost::urls::grammar::tuple_rule(
+        boost::urls::grammar::optional_rule(cfws_rule),
+        boost::urls::grammar::delim_rule('"'),
+        boost::urls::grammar::range_rule(
+            boost::urls::grammar::tuple_rule(
+                boost::urls::grammar::optional_rule(fws_rule),
+                qcontent_rule)),
+        boost::urls::grammar::optional_rule(fws_rule),
+        boost::urls::grammar::delim_rule('"'),
+        boost::urls::grammar::optional_rule(cfws_rule));
+
+/// Rule for local-part = dot-atom-text / quoted-string
+constexpr auto local_part_rule =
+    boost::urls::grammar::variant_rule(
+        dot_atom_text_rule,
+        quoted_string_rule);
+
+/// Rule for addr-spec = local-part "@" domain
+constexpr auto addr_spec_rule =
+    boost::urls::grammar::tuple_rule(
+        local_part_rule,
+        boost::urls::grammar::squelch(
+            boost::urls::grammar::delim_rule('@')),
+        domain_rule);
+
+/// Rule for to = addr-spec *("," addr-spec )
+constexpr auto to_rule =
+    boost::urls::grammar::range_rule(
+        addr_spec_rule,
+        boost::urls::grammar::tuple_rule(
+            boost::urls::grammar::squelch(
+                boost::urls::grammar::delim_rule(',')),
+            addr_spec_rule));
+
+/// Rule for hfvalue = *qchar
+constexpr auto hfvalue_rule = boost::urls::pct_encoded_rule(qchars);
+
+/// Rule for hfname = *qchar
+struct hfname_rule_t
+{
+    using value_type = boost::urls::string_view;
+
+    boost::urls::result<value_type>
+    parse(
+        char const*& it,
+        char const* end
+    ) const noexcept
+    {
+        boost::urls::string_view s(it, end);
+        it += s.size();
+        auto r = boost::urls::grammar::parse(s, hfvalue_rule);
+        if (!r)
+            return r;
+
+        // The user agent interpreting a 'mailto' URI SHOULD NOT create a
+        // message if any of the header fields are considered dangerous
+        static const boost::urls::string_view valid_k[] = {
+            "to", "subject", "keywords",
+            "cc", "body",    "in-reply-to"
+        };
+        auto match = [s](boost::urls::string_view valid_k)
+        {
+            return boost::urls::grammar::ci_is_equal(s, valid_k);
+        };
+        if (std::any_of(
+                   std::begin(valid_k),
+                   std::end(valid_k),
+                   match))
```

Comment:
What's the point of having lambdas if you are not going to put them in the argument list?

Reply:
What?

Reply:
Why not
```
    if( std::any_of( std::begin(valid_k), std::end(valid_k),
        [s]( urls::string_view valid_k )
        {
            return grammar::ci_is_equal( s, valid_k );
        };);
```
?